File: backend/src/main/java/pt/isel/rugby/model/Opponent.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table (name = "opponent")
@Data
public class Opponent implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String photo;
}

File: frontend2/Rugby/src/app/classes/opponent.ts
export class Opponent {
  constructor(
      public id?: number,
      public name?: string,
      public photo?: string,
      public file?: string
  ) {
    this.id = id ? id : 0;
    this.name = name ? name : '';
    this.photo = photo ? photo : '';
  }
}
